test/box-luatest/gh_6794_recover_nonmatching_xlogs_test.lua

```diff
@@ -0,0 +1,46 @@
+local server = require('test.luatest_helpers.server')
+local t = require('luatest')
+local g = t.group()
+local fio = require('fio')
+
+g.before_each(function()
+    g.server = server:new({alias = 'master'})
+    -- Initialize server's workdir with specially crafted snap/xlog pair.
+    os.execute(string.format('ln -s %s/* %s/',
+                             fio.abspath('./test/box-luatest/gh_6794_data'),
+                             g.server.workdir))
+end)
+
+g.after_test("test_panic_without_force_recovery", function()
+    g.server:cleanup()
+end)
+
+g.after_test("test_ignore_with_force_recovery", function()
+    g.server:drop()
+end)
+
+local mismatch_msg = "Replicaset vclock {.*} doesn't match recovered data {.*}"
+
+g.test_panic_without_force_recovery = function()
+    g.server:start({wait_for_readiness = false})
+    t.helpers.retrying({}, function()
+        local msg = "Can't proceed. " .. mismatch_msg
+        local filename = fio.pathjoin(g.server.workdir, g.server.alias..'.log')
+        assert(g.server:grep_log(msg, nil, {filename = filename}))
```

Comment:
You should use `t.assert`. Please fix here and everywhere else.

Reply:
Indeed, my bad. Thanks!

Here's the diff. Please, let me know whether you prefer small diffs per comment or a single incremental diff in reply to all comments.
```diff
@@ -26,7 +24,7 @@ g.test_panic_without_force_recovery = function()
     t.helpers.retrying({}, function()
         local msg = "Can't proceed. " .. mismatch_msg
         local filename = fio.pathjoin(g.server.workdir, g.server.alias..'.log')
-        assert(g.server:grep_log(msg, nil, {filename = filename}))
+        t.assert(g.server:grep_log(msg, nil, {filename = filename}))
     end)
 end
 
@@ -36,11 +34,11 @@ g.test_ingore_with_force_recovery = function()
     t.helpers.retrying({}, function()
         local msg = mismatch_msg .. ": ignoring, because 'force_recovery' "..
                     "configuration option is set."
-        assert(g.server:grep_log(msg))
+        t.assert(g.server:grep_log(msg))
     end)
-    assert(g.server:exec(function()
-               return box.info.signature >= 2 and
-                      box.info.status == 'running' and
-                      box.info.ro == false end),
-           "Failed to recover data")
+    t.assert(g.server:exec(function()
+                return box.info.signature >= 2 and
+                       box.info.status == 'running' and
+                       box.info.ro == false end),
+             "Failed to recover data")
 end
```